### ISSUE TITLE
Write Wingsail Subscription Logic

### ISSUE THREAD

**Issue by DFriend01** [UBCSailbot/controller#11]
### Purpose

In order for the wingsail controller node to operate, it needs to pull input data from the ROS network. This is where subscriptions come in. The idea is to "subscribe" to a ROS topic and periodically pull information produced by other ROS nodes in the network. The wingsail control node would then utilize this information for further computational purposes.

There is no subscription logic currently implemented in the wingsail node. The purpose of this issue is to subscribe to all the necessary ROS topics needed for the wingsail controller to function.

See the **Resources** section to learn more about the background tools that will be needed to successfully resolve this issue.

#### ROS Topic Subscriptions

The following ROS topics are necessary to subscribe to (based off the [wingsail controller design specification](https://ubcsailbot.atlassian.net/wiki/spaces/prjt22/pages/1785790588/New+Wingsail+Controller+Design+Specification+Old+is+at+the+bottom)):

| Topic | Input Data Provided |
| ----- | ----------------------- |
| `wind_sensors` | _**Apparent**_ Wind Speed and Direction |
| `gps` | Boat Speed and Direction |
| ?       | Angle of Wingsail |

Double-check the design specification to make sure all the input data to be considered is up to date. Take a look at the [custom interfaces repository](https://github.com/UBCSailbot/custom_interfaces) for more information on the available ROS topics and what kind of data is stored in which topics.

#### Getting Started

Modify the function `__init_subscriptions` in the `WingsailControllerNode` class located in [wingsail_ctrl_node.py](https://github.com/UBCSailbot/controller/blob/main/controller/wingsail/wingsail_ctrl_node.py).

For each subscription, a callback will need to be defined within the class. These callbacks should simply store the latest information in a class member.

For an example that was already done, check out the [physics engine node](https://github.com/UBCSailbot/boat_simulator/blob/main/boat_simulator/nodes/physics_engine/physics_engine_node.py), and in particular, look at its `__init_subscriptions` function. Ignore the `callback_group` parameter, and set `qos_profile` to 1.

#### Testing

In each subscription callback, write an info log to stdout confirming that new data has been received. You should observe this log when running the controller node. Copy and paste the output in the pull request showing the ROS info logs.

Once again, see the physics engine node as an example of how to do this.

### Resources

- [ROS2 Basic Concepts](https://docs.ros.org/en/humble/Concepts/Basic.html): Only up to (and including) ROS Topics is mandatory to resolve this issue. However, you are encouraged to reach through the entire **Basic Concepts** section to learn about how ROS2 works.
- [GitHub Development Worflow](https://ubcsailbot.github.io/docs/latest/reference/github/workflow/overview/): Read through this before starting on the issue if you have not already.
- [Sailbot Python Conventions](https://ubcsailbot.github.io/docs/latest/reference/python/conventions/): Read through this before starting on the issue if you have not already.
